Model/Component/AdminRoles.php

```diff
@@ -0,0 +1,134 @@
+
+class AdminRoles extends YamlComponentAbstract
+{
+    protected $alias = 'adminroles';
+    protected $name = 'Admin Roles';
+    protected $description = 'Component to create Admin Roles';
+
+    /**
+     * RoleFactory
+     *
+     * @var roleFactory
+     */
+    protected $roleFactory;
+
+    /**
+     * RulesFactory
+     *
+     * @var rulesFactory
+     */
+    protected $rulesFactory;
+
+    /**
+     * AdminRoles constructor.
+     * @param LoggingInterface $log
+     * @param ObjectManagerInterface $objectManager
+     * @param RoleFactory $roleFactory
+     * @param RulesFactory $rulesFactory
+     */
+    public function __construct(
+        LoggingInterface $log,
+        ObjectManagerInterface $objectManager,
+        RoleFactory $roleFactory,
+        RulesFactory $rulesFactory
+    ) {
+        parent::__construct($log, $objectManager);
+
+        $this->roleFactory = $roleFactory;
+        $this->rulesFactory = $rulesFactory;
+    }
+
+    /**
+     * @param array $data
+     * @SuppressWarnings(PHPMD)
+     */
+    protected function processData($data = null)
+    {
+
+        if (isset($data['adminroles'])) {
+            foreach ($data['adminroles'] as $role) {
+                try {
+                    if (isset($role['name'])) {
+                        $this->createAdminRole($role['name'], $role['resources']);
+                    }
+                } catch (ComponentException $e) {
+                    $this->log->logError($e->getMessage());
+                }
+            }
+        }
+    }
+
+    /**
+     * Create Admin user roles, or update them if they exist
+     *
+     * @param $roleName
+     * @param $resources
+     */
+    private function createAdminRole($roleName, $resources)
+    {
+        $role = $this->roleFactory->create();
+        $roleCount = $role->getCollection()->addFieldToFilter('role_name', $roleName)->getSize();
+
+        // Create or get existing user
+        if ($roleCount > 0) {
+            $this->log->logInfo(
+                sprintf('Admin Role "%s" creation skipped: Already exists in database', $roleName)
+            );
+
+            //Get exisiting Role
+            $role = $role->getCollection()->addFieldToFilter('role_name', $roleName)->getFirstItem();
+            $this->setResourceIds($role, $resources);
+
+            return;
+        }
+
+        $this->log->logInfo(
+            sprintf('Admin Role "%s" being created', $roleName)
+        );
+
+        $role->setRoleName($roleName)
+            ->setParentId(0)
+            ->setRoleType(RoleGroup::ROLE_TYPE)
+            ->setUserType(UserContextInterface::USER_TYPE_ADMIN)
+            ->setSortOrder(0)
+            ->save();
+
+        $this->setResourceIds($role, $resources);
+    }
+
+    /**
+     * Set ResourceIDs the Admin Role will have access to
+     *
+     * @param role
+     * @param array|null $resources
+     */
+    private function setResourceIds($role, $resources)
```

Comment:
Please add the role type in phpdoc and maybe change the 2nd param to `array $resources = null`?